### PR TITLE
feat: fixed signin validation error and improved the ux

### LINKED ISSUE #373: (BUG) Major Signin Form validation and ux bug
Bug:

The signin form has few issues with it mentioned below :
1. No proper validation was setup instead just `error signin` is shown
2. no loader or a proper loading state to show the progress of the login process 
3. lacks the good ux as well as ui 
4. it doesn't gets the value from auto-fill feature from password manager

**To Reproduce**

Steps to reproduce the behavior 1 : 
1. Go to [signup page](https://app.100xdevs.com/signin?) and input for e.g. hello.com and then click on signin
2. toast appears

Steps to reproduce the behavior 2 : 
1. Go to [signup page](https://app.100xdevs.com/signin?) and input something and then click on signin
2. No way for user to understand if the request actually went through, there should be something that shows the loading state

Steps to reproduce the behavior 4 : 
1. Go to [signup page](https://app.100xdevs.com/signin?) and if there is autofill click on signin
2. see that the values being there, it shows required error

**Expected behavior**

1.  proper validation of input when the buttons get clicked
2. loading state for better ux 
3. autofill values acceptance 
4. better ux 

**Screenshots or GIFs**
autofill error 
image
wrong validation handling 
image

**Info (please complete the following information):**
 - Browser [chrome, firefox]
 - Version [e.g. 124.0.1]

**Additional context**
i have a pr below attached that fixes all these errors

### PR DESCRIPTION
**Pull request by VineeTagarwaL-code**
### PR Fixes:
- #373

Resolves #373

- Added zod schema 
- added shadcn form 
- added loader for better ux
- fixed validation issues in the form 
- clean code 

*VALIDATION FIX* 
image

*LOADER AND UX IMPROVEMENT* 
image

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by VineeTagarwaL-code** [comment]
@hkirat please let me know if there is any changes needed

File: src/schemas/signin-schema.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { z } from 'zod';
+
+const signinSchema = z.object({
+  email: z.string().email('Invalid email address'),
```
[the remark this answers is not in the document]
Comment (VineeTagarwaL-code) [reply src/schemas/signin-schema.ts:4]: it can also have phone number as input? as it only asks for email seeing the label

**Comment by VineeTagarwaL-code** [comment]
image

remove strict validation for email and as @devsargam asked made the changes , Husky linted some files i didnt meant to changes so sorry for that @hkirat

**Review by devsargam — APPROVED** [review]

**Comment by VineeTagarwaL-code** [comment]
@hkirat ?

File: src/schemas/signin-schema.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkirat) [reply src/schemas/signin-schema.ts]: You can acually have any string

[the remark this answers is not in the document]
Comment (VineeTagarwaL-code) [reply src/schemas/signin-schema.ts]: image
it can take anything as long as the length is not 0

[the remark this answers is not in the document]
Comment (VineeTagarwaL-code) [reply src/schemas/signin-schema.ts]: should i just rename that to email or phone or username,so they know they can input any of these two

[the remark this answers is not in the document]
Comment (VineeTagarwaL-code) [reply src/schemas/signin-schema.ts]: three*
@hkirat

**Comment by hkirat** [comment]
Sorry this is a good PR but closing because of conflicts

**Pull request closed without merging**